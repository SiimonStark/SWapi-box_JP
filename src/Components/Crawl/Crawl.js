import React from 'react';
import PropTypes from 'prop-types';

const Crawl = ({film}) => {
  return (
    <div className="Crawl">
      <div className="fade"></div>
      <section className="Crawl--container">
        <article className="Crawl--content">
          <p>{film.opener}</p>
          <h3>{film.title} ({film.circa})</h3>
        </article>
      </section>
    </div>
  )
}

Crawl.propTypes = {
  film: PropTypes.object.isRequired
}

export default Crawl;