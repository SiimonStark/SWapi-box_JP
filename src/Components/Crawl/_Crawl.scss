.Crawl {
  position: relative;
  height: 100%;
  overflow: hidden;
  z-index: 3;
}

.Crawl--container {
  display: flex;
  justify-content: center;
  position: absolute;
  height: 100%;
  width: 100%;
  font-size: 250%;
  font-weight: 600;
  letter-spacing: 6px;
  line-height: 150%;
  perspective: 400px;
  text-align: center;
}

.Crawl--content {
  position: relative;
  top: 999px;
  transform-origin: 2% 100%;
  animation: crawl 20s linear;
  height: 2000px;
}

@keyframes crawl {
	0% {
    top: -100px;
    transform: rotateX(20deg)  translateZ(0);
  }
	100% { 
    top: -6000px;
    transform: rotateX(25deg) translateZ(-2500px);
  }
}