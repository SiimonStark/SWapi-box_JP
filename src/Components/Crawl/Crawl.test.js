import React from 'react';
import {shallow} from 'enzyme';
import Crawl from './Crawl';
import { mockCrawl } from "../MockTestData";

describe('Crawl', ()=>{
  let wrapper;
  beforeEach(()=>{
    wrapper = shallow(<Crawl film={mockCrawl} />);
  });

  it('should match snapshot', ()=>{
    expect(wrapper).toMatchSnapshot();
  });
})