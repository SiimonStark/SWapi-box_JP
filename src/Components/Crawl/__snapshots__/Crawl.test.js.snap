// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Crawl should match snapshot 1`] = `
<div
  className="Crawl"
>
  <div
    className="fade"
  />
  <section
    className="Crawl--container"
  >
    <article
      className="Crawl--content"
    >
      <p />
      <h4>
        (
        1977
        )
      </h4>
      <h3>
        Episode IV - A New Hope
      </h3>
    </article>
  </section>
</div>
`;
